### ISSUE TITLE
[BUG] Market Transaction Stats does not account for multiple currencies

### ISSUE THREAD

**Issue by Talha-Ak** [tfedor/AugmentedSteam#1774]
### Expected Behavior

The Market Transactions stats component should accumulate currencies separately with a total for each currency, or convert additional currencies to the account's main currency before showing total.

### Actual Behavior

(I suspect) currencies are disregarded and are all treated the same, leading to very inaccurate totals.
I also suspect the problem is exacerbated due to certain currencies using a period as a separator, while others use a comma.
![image](https://github.com/IsThereAnyDeal/AugmentedSteam/assets/30045102/456ea704-1bcd-444f-95e6-3744f518036d)
Goes without saying, but I have not earned £35 000 off the market.

### Steps to Reproduce the Problem

  1. Complete market transaction on existing currency
  2. Change Steam account to a different currency
  3. Complete market transaction on new currency

### Specifications

  - Augmented Steam Version: 2.6.0
  - Browser Version: Chrome 117.0.5938.88 (Official Build) (64-bit)

**Comment by candela97** [comment]
So when you change your wallet currency, past market transactions stay in the same currency they were processed? Can you show a screenshot that verifies this behaviour?

**Comment by Talha-Ak** [comment]
Yes, past transactions remain in the same currency they were purchased in.
The currency change doesn't happen within the same "page" of results but here are a few transactions back-to-back:
![image](https://github.com/IsThereAnyDeal/AugmentedSteam/assets/30045102/b4e2d449-57c8-42f3-bfdc-5fd4f7a2cc64)
![image](https://github.com/IsThereAnyDeal/AugmentedSteam/assets/30045102/1bb5744b-4cae-463d-81e4-93f6f87ff60f)

**Comment by candela97** [comment]
This is a tricky one. We can detect the currency by matching currency symbols, but apparently RMB and JPY share the same symbol 😅

**Comment by Talha-Ak** [comment]
Is it possible to pull in info from [purchase history](https://store.steampowered.com/account/history/)? From what I can see, it seems to render the currency code and country as a separate entry in that table every time a conversion happens.

![image](https://github.com/IsThereAnyDeal/AugmentedSteam/assets/30045102/c46db163-6420-4fad-94c9-13db0c493a6e)

**Comment by candela97** [comment]
Hmm, but don't existing listings stay in the same currency they were listed in prior to conversion?

**Comment by Talha-Ak** [comment]
Yes, they do.
What I'm thinking is (I have no idea if this is actually practical):
- If all market transactions are in the same currency (i.e. no "currency conversion" in history), keep the stats as-is currently.
- If there are currency changes (can be detected either by checking symbol or checking for "currency conversion" in history), cross-reference the date of conversion with listings made on/after that date to get the correct currency code.

That (should?) work around currencies using the same symbol.

**Comment by Talha-Ak** [comment]
> don't existing listings stay in the same currency they were listed in prior to conversion?

Sorry, maybe I misunderstood this, let me clarify.

When an account's currency is changed, all listings currently active at the time of conversion get cancelled by Steam, so it's not possible to sell an item in a different currency to what the account  currency is.

However, listings that have already passed (as I showed in the screenshots above) will show what they sold at in the currency the account/listing was previously in.

**Comment by candela97** [comment]
Oh, so they get canceled, that's good to hear. Anyway, we can probably workaround the symbol thing with some extra heuristics, since it's just 2 currencies that have this problem.

We should continue using the market history to calculate this though. Using the purchase history will make this feature cross-domain, which opens up a new can of worms.

**Comment by Talha-Ak** [comment]
Not sure if I'm stepping on anyone's toes if this is being worked on already, but I've put together a quick PoC:

![image](https://github.com/IsThereAnyDeal/AugmentedSteam/assets/30045102/0c053424-1d65-4dd4-96c8-96dc03da5e41)

This doesn't take into account the CNY/JPY issue.
Could probably check if "Cancelled by Steam" shows up between currencies to identify a change, but won't be able to determine which currency is which like that. Unless another heuristic is accessible from here?

**Comment by Talha-Ak** [comment]
Update on this issue, as there's been significant refactoring since I last looked at this. The totals now look much less egregious:

<img width="443" height="219" alt="Image" src="https://github.com/user-attachments/assets/0386f14c-e32f-4caa-9ae2-cd41ac478825" />

However, (I suspect) the amounts for each transaction are added together irrespective of the currency it's in.

I believe the PoC above is the simpler way to handle this, but it does look rather ugly.

The other way to handle it would be to get the exchange rate at the day of transaction for each purchase, convert all other currencies to the account's currency, and show a single total in the account's currency. Does not seem practical to implement to me.

Alternatively, just ignore all transactions that are not in the account's currency and note this to user in the component somewhere. This is a very specific edge case, so not sure if it's worth fixing at this point.

**Comment by graynk** [comment]
Can confirm this is still relevant. I was _very_ surprised to see this.

<img width="359" height="212" alt="Image" src="https://github.com/user-attachments/assets/21dd9c92-a297-4b39-955c-b1f69c0ce109" />
